Classes/Queue/RedisQueue.php

```diff
@@ -296,10 +295,22 @@ protected function connectClient()
     {
         $host = isset($this->clientOptions['host']) ? $this->clientOptions['host'] : '127.0.0.1';
         $port = isset($this->clientOptions['port']) ? $this->clientOptions['port'] : 6379;
+        $password = isset($this->clientOptions['password']) ? $this->clientOptions['password'] : '';
         $database = isset($this->clientOptions['database']) ? $this->clientOptions['database'] : 0;
         // The connection read timeout should be higher than the timeout for blocking operations!
         $timeout = isset($this->clientOptions['timeout']) ? $this->clientOptions['timeout'] : round($this->defaultTimeout * 1.5);
-        $connected = $this->client->connect($host, $port, $timeout) && $this->client->select($database);
+
+        $connected = $this->client->connect($host, $port, $timeout);
+        if ($connected) {
+            if ($password !== '') {
```

Comment:
Thanks for tweaking, but my main point was the additional nesting level that can be avoided by changing
```
if ($connected) {
     if ($password !== '') {
        // ...
    }
}
```
to
```
if ($connected && $password !== '') {
    // ...
}
```

Reply:
But the check for `$connected` is re-used for the `select()` call…

Reply:
There's no difference to the code paths, it's just a (tiny bit) easier to read

Reply:
Ah, there _is_ a difference. Sorry, I had to look into the file itself. the diff didn't show this

Reply:
Yeah, I was fooled myself…